charts/nd-common/templates/_authorizationpolicy.tpl

```diff
@@ -0,0 +1,52 @@
+{{- /*
+
+This function creates the an AuthorizationPolicy objects that (a) allows same-
+namespace access, and, also (b) if allowNamespaces is passed in, to allow ingress
+from them to the service.
+
+These objects are generally pretty simple, but we re-use them in a few places
+and it's nice to have one common way to make them.
+
+AuthorizationPolicies can be used in lieu of NetworkPolicies in a multi-
+cluster setup
+
+Via https://istio.io/latest/docs/concepts/security/#allow-nothing-deny-all-and-allow-all-policy:
+
+> Note the “deny by default” behavior applies only if the workload has at least one authorization
+policy with the ALLOW action.
+
+- */}}
+{{- define "nd-common.authorizationPolicy" }}
+{{- if .Values.istio.enabled }}
+---
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: {{ include "nd-common.fullname" . }}-ingress
+spec:
+  selector:
+    matchLabels:
+      {{- include "nd-common.selectorLabels" . | nindent 6 }}
+  action: ALLOW
+  rules:
+  - from:
+    - source:
+        namespaces: [{{ .Release.Namespace }}]
+{{- if .Values.ports }}
```

Comment:
You don't have to fix this - but my recommendation to make these things easier to read is to make sure to indent the `{{ if ..}}.. {{ end }}` statements so that you don't end up with this:

```
{{- end }}
{{- end }}
{{- end }}
{{- end }}
{{- end }}
```

Also, this could be turned into a single If statement.

Reply:
Yea since we're here, let's fix it (I took the multi-pattern from the networkpolicy, but single-line is more readable).

Updated.